app/controllers/sessions_controller.rb

```diff
@@ -141,22 +141,31 @@ def create
         end
       end,
       failure: -> do
-        errors = @handler_result.errors
-        lost_user = errors.any? do |error|
-          [:unknown_callback_state, :invalid_omniauth_data].include? error.code
-        end
 
-        if lost_user
-          # Something weird happened to the user's session or omniauth data and we lost their info
+        def log_and_send_to_root(dev_message, user_alert)
           Rails.logger.warn do
             oauth = request.env['omniauth.auth']
             errors = @handler_result.errors.inspect
 
-            "[SessionsCreate] Lost User on failure: " +
+            "[SessionsCreate] #{dev_message}: " +
             "OAuth data: #{oauth}; status: #{status}; errors: #{errors}"
           end
+          redirect_to root_path, alert: user_alert
+        end
```

Comment:
I suspect that defining the method inside the lambda (inside the `-> do ... end`) won't work.  If it does let me know! :-)

Reply:
Tests pass, including one for lost_user, so I think it works.